### PR TITLE
[fmt] Add CI step for black (and remove darker)

### PR DESCRIPTION
**Pull request by RMeli**
Add `black` (version `24`) to CI, and start formatting two files: `tables.py` and `due.py`. This showcases hot to incrementally add files to the `black` configuration, and how to turn off automated formatting in some portions of `tables.py`.

This PR also removes `darken`.

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4725.org.readthedocs.build/en/4725/

### CODE REVIEW

**Comment by RMeli** [comment]
https://github.com/MDAnalysis/mdanalysis/actions/runs/11143944422/job/30970132772?pr=4725#step:4:79

File: package/MDAnalysis/topology/tables.py [hunk 1]
```diff
@@ -376,28 +380,31 @@ def kv2dict(s, convertor=str):
...
+          35: 'Br', 36: 'Kr', 37: 'Rb', 38: 'Sr', 39: 'Y',  40: 'Zr', 41: 'Nb', 42: 'Mo',
+          43: 'Tc', 44: 'Ru', 45: 'Rh', 46: 'Pd', 47: 'Ag', 48: 'Cd', 49: 'In', 50: 'Sn',
+          51: 'Sb', 52: 'Te', 53: 'I',  54: 'Xe', 55: 'Cs', 56: 'Ba', 57: 'La', 58: 'Ce',
+          59: 'Pr', 60: 'Nd', 61: 'Pm', 62: 'Sm', 63: 'Eu', 64: 'Gd', 65: 'Tb', 66: 'Dy',
+          67: 'Ho', 68: 'Er', 69: 'Tm', 70: 'Yb', 71: 'Lu', 72: 'Hf', 73: 'Ta', 74: 'W',
+          75: 'Re', 76: 'Os', 77: 'Ir', 78: 'Pt', 79: 'Au', 80: 'Hg', 81: 'Tl', 82: 'Pb',
+          83: 'Bi', 84: 'Po', 85: 'At', 86: 'Rn', 87: 'Fr', 88: 'Ra', 89: 'Ac', 90: 'Th',
+          91: 'Pa', 92: 'U',  93: 'Np', 94: 'Pu', 95: 'Am', 96: 'Cm', 97: 'Bk', 98: 'Cf',
+          99: 'Es', 100: 'Fm', 101: 'Md', 102: 'No', 103: 'Lr', 104: 'Rf', 105: 'Db',
+          106: 'Sg', 107: 'Bh', 108: 'Hs', 109: 'Mt', 110: 'Ds', 111: 'Rg', 112: 'Cn',
```
Comment (RMeli) [inline package/MDAnalysis/topology/tables.py:398]: Here the automated formatting is off, but I manually removed trailing whitespaces (hence the diff).

File: package/pyproject.toml [hunk 2]
```diff
@@ -125,5 +125,10 @@ MDAnalysis = [
 [tool.black]
 line-length = 79
 target-version = ['py310', 'py311', 'py312']
-extend-exclude = '.'
+include = '''
+(
+tables\.py
+| due\.py
+)
+'''
```
Comment (RMeli) [inline package/pyproject.toml:134]: Let's start simple, with only two files. Other files or matching regexp can be added with the `|` operator.

**Comment by RMeli** [comment]
https://github.com/MDAnalysis/mdanalysis/actions/runs/11151800405/job/30996028106?pr=4725#step:4:69

**Review by IAlibay — APPROVED** [review]
lgtm!

**Comment by RMeli** [comment]
This PR now contains only 3 commits, and **should not be squash-merged**:
* ee6cbd1af543cfcb6a1ce9985d61f73725608064 enables `black` in CI and cleans up `darker`
* feca12100a2d2251fa2b4e4ad9598d9d6e00ba38 formats `due.py` and `tables.py`
* ae05db71864d8e917e8874d99c0e451b967b0f47 adds the previous commit to `.git-blame-ignore-revs`, so that `git blame` will point to the original author

The latter is the reason why the PR  **should not be squash-merged**.

**Comment by orbeckst** [comment]
@IAlibay can you please shepherd the PR as you're a reviewer and it changes the CI?

**Comment by orbeckst** [comment]
@RMeli can you please resolve conflicts?

**Comment by RMeli** [comment]
@orbeckst I don't see the conflicts, two are files being deleted... I think GitHub might have been confused by my mess with my rebasing + cherry piking commits + force pushing.

**Comment by orbeckst** [comment]
This is what it looks like to me:
image

At least I couldn't merge from the GH interface.

**Comment by RMeli** [comment]
Yes, it looks the same for me, but locally I was unable to see any conflict. I think I also rebased develop, but I'll try again later. Worst case I'll close this PR and open a new one with the commits cherry picked.

**Comment by RMeli** [comment]
I really don't know what is going on here... I just created a new branch from `develop`, cherry picked the three commits mentioned above, hard-reset this branch to the new branch, and force-pushed.

If I rebase develop, it says this is up to date. Same with merge.

**Comment by IAlibay** [comment]
@RMeli I fixed the merge conflict, but that added two commits - can you check if the commit history needs cleaning up or if it's still good to non-squash merge?

**Comment by RMeli** [comment]
Thanks @IAlibay. I see a merge commit from develop (strange, for me it did not work…), which would be better not to have in the develop branch. I’m try to squash a few commits.

**Comment by IAlibay** [comment]
> which would be better not to have in the develop branch

I mean for the odd occasion I think we can live with it.

**Comment by RMeli** [comment]
@IAlibay I rebased on `develop` and this time it seemed to work alright. It should be good to go if CI comes back green. Thanks for the help!

File: .git-blame-ignore-revs [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# black
+feca12100a2d2251fa2b4e4ad9598d9d6e00ba38
```
Comment (RMeli) [inline .git-blame-ignore-revs]:
```suggestion
a222caec815f899db77c723878054e8fd5ab88b7
```

**Comment by RMeli** [comment]
The ignored commit was wrong, I fixed that and squashed down to two commits.

**Comment by RMeli** [comment]
This workflow is not great IMO, it requires a lot of git-foo. I suggest that going forward I take note of the different `black` PRs that are squash-merged (maybe in an issue), and then periodically open a PR to add them to `.git-blame-ignore-revs`.

**Comment by orbeckst** [comment]
If we agree to use a common title for black-only commits or add a magic comment then we could (semi)automate adding to the exclusion file.

**Review by IAlibay — APPROVED** [review]

**Comment by RMeli** [comment]
Excellent suggestion @orbeckst. I suggest we use `[fmt]` as tag for formatting-related PRs.

**Pull request merged**